pkg/controller/gkenetworkparamset/gkenetworkparamset_controller_test.go

```diff
@@ -8,52 +8,59 @@ import (
 	"github.com/onsi/gomega"
 	"google.golang.org/api/compute/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/cloud-provider-gcp/crd/apis/network/v1alpha1"
 	"k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/fake"
-	networkinformers "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions"
+	gkenetworkparamset "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/network/v1alpha1"
 	"k8s.io/cloud-provider-gcp/providers/gce"
 	"k8s.io/component-base/metrics/prometheus/controllers"
+
 	"k8s.io/kubernetes/pkg/controller"
 )
 
 type testGKENetworkParamSetController struct {
-	ctx                    context.Context
-	stop                   context.CancelFunc
-	networkClient          *fake.Clientset
-	networkInformerFactory networkinformers.SharedInformerFactory
-	clusterValues          gce.TestClusterValues
-	controller             *Controller
-	metrics                *controllers.ControllerManagerMetrics
-	cloud                  *gce.Cloud
+	ctx           context.Context
+	stop          context.CancelFunc
+	networkClient *fake.Clientset
+	gnpInformer   cache.SharedIndexInformer
+	clusterValues gce.TestClusterValues
+	controller    *Controller
+	metrics       *controllers.ControllerManagerMetrics
+	cloud         *gce.Cloud
 }
 
 func setupGKENetworkParamSetController() *testGKENetworkParamSetController {
 	fakeNetworking := fake.NewSimpleClientset()
-	networkInformerFactory := networkinformers.NewSharedInformerFactory(fakeNetworking, controller.NoResyncPeriodFunc())
+	gkeNetworkParamSetInformer := gkenetworkparamset.NewGKENetworkParamSetInformer(fakeNetworking, controller.NoResyncPeriodFunc(), cache.Indexers{})
 	testClusterValues := gce.DefaultTestClusterValues()
 	fakeGCE := gce.NewFakeGCECloud(testClusterValues)
 	controller := NewGKENetworkParamSetController(
 		fakeNetworking,
-		networkInformerFactory.Networking().V1alpha1().GKENetworkParamSets(),
-		networkInformerFactory.Start,
+		gkeNetworkParamSetInformer,
 		fakeGCE,
 	)
 	metrics := controllers.NewControllerManagerMetrics("test")
+
 	return &testGKENetworkParamSetController{
-		networkClient:          fakeNetworking,
-		networkInformerFactory: networkInformerFactory,
-		clusterValues:          testClusterValues,
-		controller:             controller,
-		metrics:                metrics,
-		cloud:                  fakeGCE,
+		networkClient: fakeNetworking,
+		gnpInformer:   gkeNetworkParamSetInformer,
+		clusterValues: testClusterValues,
+		controller:    controller,
+		metrics:       metrics,
+		cloud:         fakeGCE,
 	}
 }
 
+func (testVals *testGKENetworkParamSetController) runGKENetworkParamSetController(ctx context.Context) {
+	go testVals.gnpInformer.Run(ctx.Done())
+	go testVals.controller.Run(1, ctx.Done(), testVals.metrics)
```

Comment:
this may get stuck on the wait for cache syncs ref https://github.com/kubernetes/cloud-provider-gcp/pull/417/commits/9c49aa75b3d51bccc554fa84eb10408e83bdbc4f#r1068767034

Reply:
This only gets stuck if the informer is not started during the duration of the test. I ran the test without line 55 and it gets stuck. With 55 it is fine.